### PR TITLE
api\server: Factored out UnixHttp on Windows, supported on Linux only.

### PR DESCRIPTION
**Pull request by lowenna**
First of many PRs relating to the proposal to build the docker daemon on Windows (https://github.com/docker/docker/issues/10662)

### CODE REVIEW

**Comment by jhowardmsft** [comment]
Fixed sign-off line

**Comment by crosbymichael** [comment]
LGTM

File: api/server/server_windows.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+// NewServer sets up the required Server and does protocol specific checking.
+func NewServer(proto, addr string, job *engine.Job) (Server, error) {
+	// Basic error and sanity checking
+	switch proto {
+	case "tcp":
+		return setupTcpHttp(addr, job)
+	default:
+		return nil, fmt.Errorf("Invalid protocol format. Windows only supports tcp.")
```
Comment (ahmetalpbalkan) [inline api/server/server_windows.go]: "protocol format" or just "protocol"? Also this is an Errorf without a %s. :smile: I suggest making this:

> `Invalid protocol %s. Windows only supports tcp.`

File: api/server/server_windows.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// +build windows
+
+package server
+
+import (
+	"fmt"
+	"github.com/docker/docker/engine"
```
Comment (ahmetalpbalkan) [inline api/server/server_windows.go]: non-stdlib imports are be separated with an empty line (my observation in repo in general)

Comment (cpuguy83) [inline api/server/server_windows.go]: I think that's just something go-imports does.

**Comment by cpuguy83** [comment]
LGTM

**Comment by jhowardmsft** [comment]
It's a protocol format on Linux :) But the other two comments, agreed.

**Comment by jfrazelle** [comment]
LGTM

**Comment by ahmetalpbalkan** [comment]
@jhowardmsft congrats getting the first windows daemon commit merged! :confetti_ball: :fireworks:

**Pull request merged**